controllers/growth.js

```diff
@@ -3,45 +3,208 @@
-    var start = function () {
-        var query = '',
-            from = moment().subtract(29, 'day').format('YYYY-MM-DD'), // default
-            to = moment().format('YYYY-MM-DD'), // default
-            data = util.get_data(['id'], [], req.params);
+    var data = {},
+        query_string = {},
+        from = moment().subtract(29, 'day').format('YYYY-MM-DD'), // default
+        to = moment().format('YYYY-MM-DD'); // default
+        data = util.get_data(['id'], [], req.params),
+        query_string = util.get_data([], ['from', 'to'], req.query);
 
-            query = 'SELECT * FROM channel_growth WHERE channel_id = ? ';
-            query += 'AND date BETWEEN ? AND ?';
+    function start () {
 
-            if (req.query.from && moment(req.query.from).isValid()) {
-                from = moment(req.query.from).format('YYYY-MM-DD');
-            }
+        if (query_string.from && moment(query_string.from).isValid()) {
+            from = moment(query_string.from).format('YYYY-MM-DD');
+        }
 
-            if (req.query.to && moment(req.query.to).isValid()) {
-                to =  moment(req.query.to).format('YYYY-MM-DD');
-            }
+        if (query_string.to && moment(query_string.to).isValid()) {
+            to =  moment(query_string.to).format('YYYY-MM-DD');
+        }
+
+        if (query_string.to && query_string.from) {
+            get_stat(null, []);
+        }
+
+        mysql.open(config.DB)
+            .query('SELECT channel_id, title, network, linked_date FROM mcn_channels '
+                   + 'WHERE channel_id = ?',
+                   [data.id],
+                   get_stat
+            )
+            .end();
+    }
+
+    function get_stat (err, result) {
+        if (err) {
+            return next(err);
+        }
+
+        if (result.length) {
+            from = moment(result[0].linked_date).format('YYYY-MM-DD');
+            to = moment(result[0].linked_date).add(2, 'month').format('YYYY-MM-DD');
+        }
+
+        mysql.open(config.DB)
+            .query(
+                'SELECT * FROM channel_stats WHERE channel_id = ? '
+                + 'AND insert_date BETWEEN ? AND ? ORDER BY insert_date',
+                [data.id, from, to],
+                send_response
+            )
+            .end();
+    }
+
+    function send_response (err, result) {
+        if (err) {
+            return next(err);
+        }
+
+        if (!result.length) {
+            return res.warn(404, 'No channel statistics');
+        }
+
+        res.send(result);
+    }
+
+    start();
+};
+
+
+
+exports.get_grouped_channel = function (req, res, next) {
+    var date = moment(),
+        start_month = date.subtract(3, 'month').format('YYYY-MM'),
+        end_month = date,
+        data = util.get_data([], ['month', 'network',
+                             'duration'], req.query),
+        channels = [],
+        cache_key,
+        cache = new memcached(config.memcached.nodes);
```

Comment:
move cache declaration to the top so you have the uninitialized variable at the bottom